tests/rptest/tests/e2e_topic_recovery_test.py

```diff
@@ -169,3 +184,112 @@ def s3_has_all_data():
         self.logger.info(f"Consumer read {consume_total} messages")
         assert produce_acked >= num_messages
         assert consume_valid >= produce_acked, f"produced {produce_acked}, consumed {consume_valid}"
+
+    @cluster(num_nodes=4)
+    @matrix(recovery_overrides=[{}, {
+        'retention.bytes': 1024,
+        'redpanda.remote.write': True,
+        'redpanda.remote.read': True,
+    }])
+    def test_restore_with_aborted_tx(self, recovery_overrides):
+        """Produce data using transactions includin some percentage of aborted
+        transactions. Run recovery and make sure that record batches generated
+        by aborted transactions are not visible.
+        This should work because rm_stm will rebuild its snapshot right after
+        recovery. The SI will also supports aborted transctions via tx manifests.
+
+        The test has two variants:
+        - the partition is downloaded fully;
+        - the partition is downloaded partially and SI is used to read from the start;
+        """
+        if self.debug_mode:
+            self.logger.info(
+                "Skipping test in debug mode (requires release build)")
+            return
+
+        producer = Producer(self.redpanda.brokers(), "topic-recovery-tx-test",
+                            self.logger)
+
+        def done():
+            for _ in range(100):
+                try:
+                    producer.produce(self.topic)
+                except ck.KafkaException as err:
+                    self.logger.warn(f"producer error: {err}")
+                    producer.reconnect()
+            self.logger.info("producer iteration complete")
+            topic_partitions = segments_count(self.redpanda,
+                                              self.topic,
+                                              partition_idx=0)
+            partitions = []
+            for p in topic_partitions:
+                partitions.append(p >= 10)
+            return all(partitions)
+
+        wait_until(done,
+                   timeout_sec=120,
+                   backoff_sec=1,
+                   err_msg="producing failed")
+
+        assert producer.num_aborted > 0
```

Comment:
is this guaranteed to be true if we are relying on rng?

Reply:
We're relying on rng but the probability of the assertion is effectively zero.